HW2_esmythe_SQL.sql

```diff
@@ -0,0 +1,87 @@
+/*
+  Created by Ed Smythe Sept 2020
+  BDA 696 @ SDSU
+*/
+
+-- make unique batter list
+drop table if exists player_bat_avg;
+create table player_bat_avg as
+select unique batter from atbat_r order by batter;
+
+-- make table to pull performance data from
+drop table if exists player_bat_avg_all;
+create table player_bat_avg_all as
+select batter_counts.batter, hit, game.game_id,  game.local_date from batter_counts 
+join game on batter_counts.game_id = game.game_id  
+order by batter, game_id;
+
+-- Make Report Table
+drop table if exists REPORT;
+create table REPORT as select distinct batter from player_bat_avg_all where batter = '400284';
+alter table REPORT 
+  add historic float
+    after batter,
+  add annual float 
+    after historic,
+  add rolling float
+   after annual;
+   
+-- get specific batter performance
+drop table if exists hist_calc;
+create table hist_calc as
+select batter, count(case when hit = 0 then 1 end) as hit, count(hit) as bat
+from player_bat_avg_all
+where batter = '400284';-- chg
+/*this will turn into a user entered value in future iteration*/
+
+-- Make report table
+alter table hist_calc 
+  add hist_avg float;
+    
+update hist_calc set hist_avg = (hit/bat) where batter = '400284'; -- chg
+/*this will turn into a user entered value in future iteration*/
+
+-- send results to report table
+update REPORT , hist_calc
+set REPORT.historic = hist_calc.hist_avg
+where REPORT.batter = hist_calc.batter;
+
+-- Make annual table
+drop table if exists annual_calc;
+create table annual_calc as
+select batter, count(case when hit = 0 then 1 end) as hit_an , count(hit) as bat_an
+from player_bat_avg_all
+where batter = '400284' and local_date like'2011%';-- chg
+/* these will turn into a user entered values in future iteration
+   select max(local_date), min(local_date) from game
+   Min = 2007-02-26 13:05:00 and max = 2012-06-28 22:15:00	*/
+
+alter table annual_calc 
+  add annual float
+    after bat_an;
+
+update annual_calc set annual_calc.annual = (hit_an/bat_an);
+
+-- send results to report table
+update REPORT , annual_calc
+set REPORT.annual = annual_calc.annual
+where REPORT.batter = annual_calc.batter;
+
+-- Make rolling table
+drop table if exists rolling_calc;

```

Comment:
This doesn't look right (I sent out an announcement to the class about this too).
A "rolling" value would imply that you have the value for all the dates, not just user specified. Don't fret too much about not having it done by Tuesday (I'm going to give some extra time), but the "rolling" part is missing. here.

Reply:
So here now that i have it "working" i plan on making a aolumn for all unique dates of games and then calculate the average for each player (row) over the last 100 days based on the column. Is there ea better way that you recommend?

Reply:
You can do it that way, but I think the easier way to get it working would be to get a list of all the games and players in those games. Then, for each of those rows, (playerid, game_id) join the table to itself (which is totally valid) and then put a restriction on that join on the dates (you may have to rope in a date table). 

That way, instead writing a for loop or something, you we get everything using simple select statements.

Reply:
PS - I was talking to some students today (they contacted me to setup office hours) and they told me they have invited you to join their PR team (a team of 3!). I think that would be good. 

I'm always available for questions if you like (and I'm trying my best to be available to everyone), but it would be good to get a back and forth going with your peers. I'm very restricted to look at stuff during the week and basically devote my whole weekends to this class.

Because of that, I may not be able to give you the feedback you need in a timely manner. 

I could phrase it another way too: "It's not you, it's me" :-D 
j/k
---------------------
PPS - I love that PDF how-to document you posted! I've said it before, but I'll say it again: persistence pays off in this field, and you have plenty of it!

Reply:
AH i was looking at that method last night based on reviewing your class notes. 

I received an email from Nika about it. we will make ti work.

Thank you for your efforts so far.

Thank you for the compliments about my persistence.

Reply:
went with a self join but not sure if it is stuck or needs more time to complete. Left running for 30 min without completion.